jetty-websocket/websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/internal/AbstractMessageHandler.java

```diff
@@ -0,0 +1,190 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.websocket.core.internal;
+
+import java.nio.ByteBuffer;
+
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.websocket.core.CloseStatus;
+import org.eclipse.jetty.websocket.core.CoreSession;
+import org.eclipse.jetty.websocket.core.Frame;
+import org.eclipse.jetty.websocket.core.FrameHandler;
+import org.eclipse.jetty.websocket.core.OpCode;
+import org.eclipse.jetty.websocket.core.exception.BadPayloadException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A utility implementation of FrameHandler that de-fragments text frames and binary frames into a whole messages before
+ * calling {@link #onText(String, Callback)} or {@link #onBinary(ByteBuffer, Callback)}.
+ * This is a demanding frame handler so
+ * Flow control is by default automatic, but an implementation
+ * may extend {@link #isDemanding()} to return true and then explicityly control
+ * demand with calls to {@link CoreSession#demand(long)}
+ */
+public abstract class AbstractMessageHandler implements FrameHandler
```

Comment:
This class is still `internal`, but maybe it'll be useful to expose it?

For example, to use a different accumulation method, or to asynchronously proxy frames to other components (hence no accumulation needed), I would need to extend from this class to reuse most/all of its code, but I can't if it's internal.

Reply:
I have moved this outside internal package to `org.eclipse.jetty.websocket.core.util`.